src/MySTMarkdownCell.tsx

```diff
@@ -17,11 +17,28 @@ import { useParse } from 'myst-to-react';
 import { parseContent } from './myst';
 import { IMySTMarkdownCell } from './types';
 import { linkFactory } from './links';
+import { selectAll } from 'unist-util-select';
+
+import { PromiseDelegate } from '@lumino/coreutils';
+import { metadataSection, IUserExpressionMetadata } from './metadata';
+import { CellMetadataProvider } from './cellMetadataProvider';
+import { IRenderMimeRegistry } from '@jupyterlab/rendermime';
 
 export class MySTMarkdownCell
   extends MarkdownCell
   implements IMySTMarkdownCell
 {
+  private _doneRendering = new PromiseDelegate<void>();
```

Comment:
There is now a promise for when the initial mdast parse is complete for the cell, then this can trigger the inline execution.